crates/iota/src/name_commands.rs

```diff
@@ -1311,6 +1790,29 @@ async fn fetch_renewal_config(context: &mut WalletContext) -> anyhow::Result<Ren
     Ok(entry.renewal_config)
 }
 
+async fn handle_transaction_result<Fun, F>(
+    res: IotaClientCommandResult,
+    fun: Fun,
+) -> anyhow::Result<NameCommandResult>
+where
+    Fun: FnOnce(IotaTransactionBlockResponse) -> F,
+    F: futures::Future<Output = anyhow::Result<NameCommandResult>>,
+{
+    Ok(
+        if let IotaClientCommandResult::TransactionBlock(res) = res {
+            if !res.errors.is_empty() {
+                for err in res.errors {
+                    println!("{err}");
+                }
+                bail!("transaction failed");
```

Comment:
Shouldn't this rather be before the errors with `:`?

Reply:
Yeah, I was thinking the same, but wasn't sure the best way to do it. I'll think about it a bit